hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java

```diff
@@ -134,7 +135,7 @@ protected synchronized void chore() {
       loadRegionsFromInMemoryState();
       loadRegionsFromRSReport();
       try {
-        loadRegionsFromFS();
+        loadRegionsFromFS(scanForMergedParentRegions());
```

Comment:
would it make sense to make it a bit more generic and have "loadRegionsFromMeta()" function (similar to loadRegionsFromInMemoryState/loadRegionsFromRSReport) as then you'd have another source to compare against - hbase:meta. Then loadRegionsFromFS() would check against that state to see if a region is merged or not.

Reply:
Yeah, that is the idea. scanForMergedParentRegions() loads regions from meta. For merge, it is a bit special, the parent regions are deleted from meta already, the only bit left is the merge qualifers in the child region.

Reply:
Does InMemoryState not have the needed merge info in it? If not, maybe it should.

The CatalogJanitor is what manages when merge references are let go so this scan of meta is probably necessary.

To the @timoha point, are there other places in hbckchore where we need currrent picture of hbase:meta?

Reply:
@saintstack Yeah the in-memory state/meta row for merged parents are let go at early state of MergeRegionsProcedures. 

@timoha Per Stack's comments about the source of truth is the in-memory database (meta/procedure store are ways to recover in-memory databse since they are persistent). 

At this moment, there is no other usage of regions from meta, the merged parents info is a special case, they are columns from the child region, but the only source of truth for merged parents. We can maintain the in-memory hashset for merged parent if meta scan is too costly, which can be addressed later. In case of the future requirements, scanForMergedParentRegions() can be modified to get more info from meta.